lib/metadata_json_lint.rb

```diff
@@ -40,6 +42,10 @@ def run
         options[:fail_on_warnings] = v
       end
 
+      opts.on('--[no-]strict-puppet-version', "Fail on strict Puppet Version check based on current supported Puppet versions. Defaults to '#{options[:strict_puppet_version]}'.") do |v|
+        options[:strict_puppet_version] = v
```

Comment:
Given some concerns about changing option names and moving this to a plugin, perhaps this should be marked `[Experimental]` or with similar text?

Reply:
I don't think we're planning on moving this option specifically to a plugin. This is designed to be the puppet version equivalent to `--strict-dependencies`. Simply enforce puppet version requirements based on supported versions and semver requirements. An on/off option that, like `--strict-dependencies` is off by default.

Additional tests, like being able to specify your own version minimum and test against that, would be a plugin as it's outside the simple scope of the base tool.